onnxruntime/core/providers/cpu/nn/conv_attributes.h

```diff
@@ -136,14 +136,14 @@ struct ConvAttributes {
                              " W: ", W->Shape().ToString().c_str());
     }
 
-    if (C != W->Shape()[1] * group) {
+    if (-1 != group && C != W->Shape()[1] * group) {
```

Comment:
ensure `W->Shape()[1] != 0` and calculate real `group` value and check the 2 conditions, do not skip them.

this implicitly includes checking for `C % W->Shape()[1] == 0` and `M % (C % W->Shape()[1]) == 0`